cmd/ctr/commands/content/content.go

```diff
@@ -539,14 +539,19 @@ func edit(context *cli.Context, rd io.Reader) (io.ReadCloser, error) {
 		tmp.Close()
 		return nil, err
 	}
-
-	if _, err := tmp.Seek(0, io.SeekStart); err != nil {
-		tmp.Close()
+	// need reopen tmp cause it is edited in another process
+	tmp.Close()
```

Comment:
I think the Close should be called after Copy before the editor run.

Reply:
the content only update when Close() after editor run 
in my host, the filesystem type is ext4
`zhanghe@zhanghe:~/go/src/github.com/containerd/containerd$ sudo bin/ctr content edit sha256:a5d93b6db095ac1ddfcb976df7a6cebac94a3826d21dfefa42ad43b58b159916 -editor /usr/bin/vim
ctr: commit failed: blob with expected digest sha256:a5d93b6db095ac1ddfcb976df7a6cebac94a3826d21dfefa42ad43b58b159916 exists: already exists`

Reply:
Actually it is not. Please check my last comment. In ext4 case, the editor will create a new file and rename to override the old one. The ctr holds deleted fd.

I was saying that the tmp.Close should be called after Line 526 so that we don't need to Close in every error check condition.

_, err = io.Copy(tmp, rd)
tmp.Close()
if err != {
}

...